cmd/kill.go

```diff
@@ -0,0 +1,103 @@
+package cmd
+
+import (
+	"log"
+
+	"github.com/Originate/git-town/lib/config"
+	"github.com/Originate/git-town/lib/git"
+	"github.com/Originate/git-town/lib/prompt"
+	"github.com/Originate/git-town/lib/steps"
+	"github.com/spf13/cobra"
+)
+
+type KillFlags struct {
+	Undo bool
+}
+
+type KillConfig struct {
+	InitialBranch       string
+	IsTargetBranchLocal bool
+	TargetBranch        string
+}
+
+var killFlags KillFlags
+
+var killCommand = &cobra.Command{
+	Use:   "kill [<branch>]",
+	Short: "Removes an obsolete feature branch",
+	Long:  "Removes an obsolete feature branch",
+	Run: func(cmd *cobra.Command, args []string) {
+		steps.Run(steps.RunOptions{
+			CanSkip:              func() bool { return false },
+			Command:              "kill",
+			IsAbort:              false,
+			IsContinue:           false,
+			IsSkip:               false,
+			IsUndo:               killFlags.Undo,
+			SkipMessageGenerator: func() string { return "" },
+			StepListGenerator: func() steps.StepList {
+				if len(args) > 1 {
+					log.Fatal(cmd.UsageTemplate())
+				}
+				killConfig := checkKillPreconditions(args)
+				return getKillStepList(killConfig)
+			},
+		})
+	},
+}
+
+func checkKillPreconditions(args []string) (result KillConfig) {
+	result.InitialBranch = git.GetCurrentBranchName()
+
+	if len(args) == 0 {
+		result.TargetBranch = result.InitialBranch
+	} else {
+		result.TargetBranch = args[0]
+	}
+
+	config.EnsureIsFeatureBranch(result.TargetBranch, "You can only kill feature branches.")
+
+	result.IsTargetBranchLocal = git.HasLocalBranch(result.TargetBranch)
```

Comment:
The kill command always has a target branch though and the point of the variable is whether or not it is local.